spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioClientConnectionFactory.java

```diff
@@ -113,6 +115,11 @@ protected TcpConnectionSupport buildNewConnection() {
 			return wrappedConnection;
 		}
 		catch (IOException e) {
+			try {
+				socketChannel.close();
```

Comment:
Don't we need to check if `socketChannel != null` first?

Reply:
Ah right, I missed the null check. Thanks for catching that!